appservice/src/startStreamingLogs.ts

```diff
@@ -55,21 +56,30 @@ export async function startStreamingLogs(client: ISimplifiedSiteClient, verifyLo
                     timerId = setInterval(async () => await pingFunctionApp(client), 60 * 1000);
                 }
 
-                await new Promise<void>((onLogStreamEnded: () => void, reject: (err: Error) => void): void => {
-                    const logsRequest: request.Request = request(`${client.kuduUrl}/api/logstream/${logsPath}`, {
-                        auth: {
-                            user: creds.publishingUserName,
-                            password: nonNullProp(creds, 'publishingPassword')
-                        },
-                        headers: {
-                            'User-Agent': appendExtensionUserAgent()
-                        }
-                    });
+                const genericClient: ServiceClient = await createGenericClient(new BasicAuthenticationCredentials(creds.publishingUserName, nonNullProp(creds, 'publishingPassword')));
 
+                const abortController: AbortController = new AbortController();
+
+                // Due to https://github.com/Azure/ms-rest-js/issues/393 we need to use the WebResource constructor
+                // When that is fixed this can be simplified to just an options object passed to `genericClient.sendRequest`
+                const request = new WebResource(
```

Comment:
I tried it out with Azure Functions, and it definitely works initially but seems to quit working after a bit. Maybe a timeout?